### PR TITLE
Add workflow to build go tooling on tag push

### PR DESCRIPTION
**Pull request by adamruzicka**

### CODE REVIEW

**Comment by adamruzicka** [comment]
It generates releases like this one https://github.com/adamruzicka/dynflow/releases/tag/v1.6.3

File: .github/workflows/release.yml [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+        with:
+          tagRegex: "v(.*)"  # Optional. Returns specified group text as tag name. Full tag string is returned if regex is not defined.
+          tagRegexGroup: 1 # Optional. Default is 1.
+      - name: Checkout the repository
+        uses: actions/checkout@v2
+      - name: Generate build files
+        run: |
+          mkdir -p dist
+          cd extras/expand
+          go build -o ../../dist/dynflow-expand-${{ steps.tagName.outputs.tag }}-x86_64
```
Comment (ezr-ondrej) [inline .github/workflows/release.yml:27]: I'd rather see VERSION used here as well, or not used in the other task as well, but its nit :)

**Review by ezr-ondrej — APPROVED** [review]
The GH action looks good and I like removing code :thinking:

File: .github/workflows/release.yml (hunk 1 above)
Comment (adamruzicka) [reply .github/workflows/release.yml]: Done

**Comment by adamruzicka** [comment]
I addressed the last comment and squashed the commits a bit

**Review by ezr-ondrej — APPROVED** [review]
Thanks @adamruzicka ! :+1:

**Pull request merged**